src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2474,7 +2484,13 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           if (result0.nonEmpty) checkPure(result0, supple = true)
         }
 
-        treeCopy.Block(block, statsTyped, expr1)
+        // Remove ValDef for right-associative by-value operator desugaring which has been inlined into expr1
+        val statsTyped2 = statsTyped match {
+          case (vd: ValDef) :: Nil if inlinedRightAssocValDefs contains vd.symbol => Nil
+          case _ => statsTyped
+        }
+
+        treeCopy.Block(block, statsTyped2, expr1)
```

Comment:
`typedBlock` could be changed to return a `Tree` but I didn't want to add unnecessary complications. The empty blocks don't seem to cause any problem.